commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentMethodGroupRelLocalServiceImpl.java

```diff
@@ -306,6 +330,77 @@ public CommercePaymentMethodGroupRel setActive(
 
 		commercePaymentMethodGroupRel.setActive(active);
 
+		if (!active) {
+			Map<String, CommercePaymentMethod> commercePaymentMethodsMap =
+				_commercePaymentMethodRegistry.getCommercePaymentMethods();
+
+			List<CommercePaymentMethod> recurringCommercePaymentMethods =
+				new ArrayList<>();
+
+			for (CommercePaymentMethod commercePaymentMethod :
+					commercePaymentMethodsMap.values()) {
+
+				if (commercePaymentMethod.isProcessRecurringEnabled()) {
+					recurringCommercePaymentMethods.add(commercePaymentMethod);
+				}
+			}
+
+			if (recurringCommercePaymentMethods.size() == 1) {
+				CommercePaymentMethod lastRecurringCommercePaymentMethod =
+					recurringCommercePaymentMethods.get(0);
+
+				String lastRecurringCommercePaymentMethodKey =
+					lastRecurringCommercePaymentMethod.getKey();
+
+				if (!lastRecurringCommercePaymentMethodKey.equals(
+						commercePaymentMethodGroupRel.getEngineKey())) {
+
+					return commercePaymentMethodGroupRelPersistence.update(
+						commercePaymentMethodGroupRel);
+				}
+
+				SearchContext searchContext = new SearchContext();
```

Comment:
@alecsloan why we are using search framewrok and not a simpler service call ?

Reply:
I'll add a finder for getting definitions by subscription enabled and channelGroupId

Reply:
After changing to a finder and testing I remembered that I was using the search because we need to look at all the products in the current channel. With the finder we would have to look in specific catalogs.

Reply:
Removing that custom message also. Instead of telling the admin how many products it will disable, the message will just be general.